src/pattern-library/components/patterns/SpinnerComponents.js

```diff
@@ -25,6 +30,22 @@ export default function SpinnerComponents() {
           </Library.Demo>
         </Library.Example>
       </Library.Pattern>
+
+      <Library.Pattern title="Full-Screen Spinner">
+        <Library.Example>
+          <p>
+            A component that renders a full-screen spinner over an overlay. Note
+            that this page has to be reloaded to clear the full-screen spinner
+            after showing it.
```

Comment:
It would be nice to have a way of dismissing the spinner without a full page reload. Maybe something like:

```diff
diff --git a/src/pattern-library/components/patterns/SpinnerComponents.js b/src/pattern-library/components/patterns/SpinnerComponents.js
index ab244bb..17f9f84 100644
--- a/src/pattern-library/components/patterns/SpinnerComponents.js
+++ b/src/pattern-library/components/patterns/SpinnerComponents.js
@@ -7,6 +7,7 @@ export default function SpinnerComponents() {
   const [fullScreenSpinnerVisible, setFullScreenSpinnerVisible] = useState(
     false
   );
+  const hideSpinner = () => setFullScreenSpinnerVisible(false);
   return (
     <Library.Page title="Spinner">
       <p>
@@ -42,7 +43,13 @@ export default function SpinnerComponents() {
             <LabeledButton onClick={() => setFullScreenSpinnerVisible(true)}>
               Show Full-Screen Spinner
             </LabeledButton>
-            {fullScreenSpinnerVisible && <FullScreenSpinner />}
+            {fullScreenSpinnerVisible && (
+              // Suppress a11y errors
+              // eslint-disable-next-line
+              <div onClick={hideSpinner}>
+                <FullScreenSpinner />
+              </div>
+            )}
           </Library.Demo>
         </Library.Example>
       </Library.Pattern>
```

(If there is an easy fix for the a11y complaint great, but otherwise I think we can accept it as a known issue for this developer-facing tool).

Reply:
@robertknight I've pushed a commit that uses `use-element-should-close` hook to make it so that you can close the spinner with `ESC` in both pattern library examples.